lib/spack/spack/test/cmd/spec.py

```diff
@@ -55,6 +55,16 @@ def test_spec_concretizer_args(mutable_config, mutable_database):
     assert h in output
 
 
+def test_spec_parse_arg_quoting():
+    """Assert that certain spec constructions work when provided as arguments."""
+    # Verify that we can provide multiple key=value variants to multiple separate
+    # packages within a spec string.
+    output = spec('multivalue-variant fee=barbaz ^ a foobar=baz')
+
+    assert 'fee=barbaz' in output
+    assert 'foobar=baz' in output
+
```

Comment:
I believe that this test succeeds even without the changes in this PR, can you double-check on that?

Reply:
You have truly fantastic timing @cosmicexplorer.  Would you mind testing: 
- `spack spec ‘gcc@4.9.3 cflags=“-Os -pipe” cxxflags=“-Os -pipe”`
- `spack spec ‘gcc@4.9.3 cflags=-Os -pipe cxxflags=-Os -pipe`
and see what comes out?

This should definitely help, but I think we may still have an over-eager glob on options with a VAL token.

Reply:
@becker33:
> I believe that this test succeeds even without the changes in this PR, can you double-check on that?

I orchestrated the commits so that the test is added before the fix, and this test indeed fails without the fix:
```
spack.variant.InvalidVariantValueError: invalid values for variant "fee" in package "multivalue-variant": ['barbaz ^ a foobar=baz']
```

@trws:
```bash
> spack spec 'gcc@4.9.3 cflags="-Os -pipe" cxxflags="-Os -pipe"'
...
gcc@4.9.3%gcc@11.2.1 cflags="-Os -pipe" cxxflags="-Os -pipe" ~binutils+bootstrap~graphite~nvptx~piclibs~strip languages=c,c++,fortran patches=0689c26,27e9829,7112abb,f5643d9,f73de87 arch=linux-alpine3-zen3
...
> spack spec 'gcc@4.9.3 cflags=-Os -pipe cxxflags=-Os -pipe'
==> Error: Cannot specify variant "pipe" twice
```

Regarding the second command line with `cflags=-Os -pipe`: that one is currently accepted by spack on `develop`, so while I think this change should still be made, it may break some users. How should we notify users about this?

Additionally, we can verify the issue from https://github.com/spack/spack/pull/29282#issuecomment-1057496052 is fixed with the `--yaml` output:

```
> spack spec --yaml 'gh cflags="-Os -pipe" cxxflags="-Os -pipe"' | yq
...
        cflags:
          - -Os
          - -pipe
...
> git checkout develop
> spack spec --yaml 'gh cflags="-Os -pipe" cxxflags="-Os -pipe"' | yq
...
        cflags:
          - '"-Os'
          - -pipe" cxxflags="-Os
          - -pipe"
...
```

So it looks like that would absolutely be a good one to include as a new test.